### ISSUE TITLE
Nativescript 1.1: crash when instancing a custom class

### ISSUE THREAD

**Issue by Zylann** [GodotNativeTools/godot-cpp#96]
Tested with the nativescript 1.1 branch:

I registered the following class:

```cpp
class Test : public godot::Reference {
	GODOT_CLASS(Test, godot::Reference)
public:
	Test() {
		godot::Godot::print("Construct Test");
	}

	~Test() {
		godot::Godot::print("Destroy Test");
	}

	void foo() {
		godot::Godot::print("Hello world!");
	}

	static void _register_methods() {
		godot::register_method("foo", &Test::foo);
	}
};
```

Which I instanced from GDScript, but it caused a crash just after the print "Construct Test":

![image](https://user-images.githubusercontent.com/1311555/36344877-2b3d6c5c-1421-11e8-89d4-d20734ef36eb.png)

mb is 0x0.

**Comment by Zylann** [comment]
My bad I didnt notice the SConstruct was wrong

**Comment by Zylann** [comment]
Ah, sorry but... the SConstruct was right, I had all correct arguments hidden in QtCreator's build command line :D

So... I reopen the issue, rebuilding didn't fix it.

**Comment by and3rson** [comment]
Sorry for bumping, but I can still reproduce this as well. Instantiating custom class with exact same circumstances as OP mentioned also crashes the process:

```
[1] /usr/lib/libc.so.6(+0x37e00) [0x7f8e82689e00] (??:0)
[2] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x90c100] (<artificial>:?)
[3] godot::___godot_icall_bool(godot_method_bind*, godot::Object const*) (??:0)
[4] godot::Reference::init_ref() (??:0)
[5] godot::Ref<godot::GDNativeLibrary>::ref_pointer(godot::GDNativeLibrary*) (??:0)
[6] godot::Ref<godot::GDNativeLibrary>::Ref(godot::GDNativeLibrary*) (??:0)
[7] godot::Foo::_new() (??:0)
[8] godot::ForeignLibrary::open(godot::String) (??:0)
[9] void godot::_WrappedMethod<godot::ForeignLibrary, godot::Foo*, godot::String>::apply<0>(godot::Variant*, godot::ForeignLibrary*, godot::Variant**, godot::__Sequence<0>) (??:0)                                                         
[10] godot_variant godot::__wrapped_method<godot::ForeignLibrary, godot::Foo*, godot::String>(void*, void*, void*, int, godot_variant**) (??:0)                                                                                             
[11] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0xc87957] (<artificial>:?)
[12] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x217466f] (??:?)
[13] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x2471b4e] (??:?)
[14] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x8f4702] (??:?)
[15] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0xa49835] (<artificial>:?)
[16] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x16b96d5] (??:?)
[17] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x17f0898] (<artificial>:?)
[18] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x2129a84] (??:?)
[19] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x163d5cc] (<artificial>:?)
[20] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x163d523] (<artificial>:?)
[21] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x16b9057] (<artificial>:?)
[22] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x16b9166] (??:?)
[23] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x7b28cb] (??:?)
[24] /usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7f8e82676223] (??:0)
[25] /home/anderson/downloads/Godot_v3.1-stable_x11.64() [0x7bfc4e] (??:?)
```

**Comment by karroffel** [comment]
Does your class have the `_init` function defined?

This is technically an error inside of Godot, since `_init` is registered as a virtual function of `Object`, but it doesn't actually exist **anywhere**.

The C++ bindings use the `_init` function as a replacement of a constructor, since at the time of running the constructor the "script owner" is not set yet, so any calls to the `self` will fail. Inside of `_init` the owner is set.

**TL;DR**: do you have a `void _init()` function in your class? It is necessary and will crash if it doens't exist.
